coreUtilities/src/main/java/org/opensearch/migrations/tracing/RootOtelContext.java

```diff
@@ -177,7 +177,6 @@ private static Span buildSpanWithParent(SpanBuilder builder, Span parentSpan, St
         String spanName,
         Stream<Span> linkedSpans
     ) {
-        assert forScope.getCurrentSpan() == null;
```

Comment:
why are we removing asserts?

Reply:
For what it's worth, I've never been happy w/ how I initialize the spans this way - but this assert has fired a number of times during development.  It's been super valuable in the past.  If you'd like to explain it more, please feel free to w/ the : "..." syntax.

Reply:
This seems redundant because the getter is annotated with NonNull

```
    @NonNull
    Span getCurrentSpan();
```

Reply:
Good catch.  I'm fine removing it.